types/seat/wlr_seat_pointer.c

```diff
@@ -304,6 +305,28 @@ void wlr_seat_pointer_notify_motion(struct wlr_seat *wlr_seat, uint32_t time,
 	grab->interface->motion(grab, time, sx, sy);
 }
 
+void wlr_seat_pointer_notify_relative_motion(struct wlr_seat *wlr_seat,
+		uint64_t time, double dx, double dy,
+		double dx_unaccel, double dy_unaccel) {
+	struct wlr_seat_client *client = wlr_seat->pointer_state.focused_client;
+	if (client == NULL) {
+		return;
+	}
+
+	struct wl_resource *resource;
+	wl_resource_for_each(resource, &client->relative_pointers) {
+		struct wlr_relative_pointer_v1 *relative_pointer =
+			wl_resource_get_user_data(resource);
```

Comment:
There should be a safe helper function that asserts `wl_resource_is_instance_of` instead of directly using `wl_resource_get_user_data`